static/js/settings_profile_fields.js

```diff
@@ -50,13 +68,7 @@ function create_profile_field(e) {
 
     settings_ui.do_settings_change(channel.post, "/json/realm/profile_fields", $(this).serialize(),
                                    $('#admin-profile-field-status').expectOne());
-}
-
-function create_choice_row(container, add_delete_button) {
-    var context = {};
-    context.add_delete_button = add_delete_button;
-    var row = templates.render("profile-field-choice", context);
-    $(container).append(row);
+    clear_form_data();
```

Comment:
I think we only want to call this in the success callback (may need to plumb into `do_settings_change`); otherwise, if there's an error, you lose your work.